doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -287,123 +279,98 @@ automatically created.
 
-      This command adds :n:`simple apply @qualid` to the hint list with the head
-      symbol of the type of :n:`@qualid`. The cost of that hint is the number of
-      subgoals generated by :n:`simple apply @qualid` or :n:`@natural` if specified. The
+      .. todo PR: wordy paragraph, maybe should have a little table instead of writing
+         out all the conditions in words?
+
+      The first form adds :tacn:`simple apply` :n:`@qualid` to the hint list for each :n:`@qualid`
+      with the head symbol of the type of :n:`@qualid`. The cost of the hint is the number of
+      subgoals generated by :tacn:`simple apply` :n:`@qualid` or, if specified, :n:`@natural`. The
       associated :n:`@pattern` is inferred from the conclusion of the type of
-      :n:`@qualid` or the given :n:`@pattern` if specified. In case the inferred type
+      :n:`@qualid` or, if specified, the given :n:`@pattern`. If the inferred type
       of :n:`@qualid` does not start with a product the tactic added in the hint list
-      is :n:`exact @qualid`. In case this type can however be reduced to a type
-      starting with a product, the tactic :n:`simple apply @qualid` is also stored in
+      is :tacn:`exact` :n:`@qualid`. However, in case this type can be reduced to a type
+      starting with a product, the tactic :tacn:`simple apply` :n:`@qualid` is also stored in
       the hints list. If the inferred type of :n:`@qualid` contains a dependent
       quantification on a variable which occurs only in the premisses of the type
       and not in its conclusion, no instance could be inferred for the variable by
       unification with the goal. In this case, the hint is added to the hint list
-      of :tacn:`eauto` instead of the hint list of auto and a warning is printed. A
-      typical example of a hint that is used only by :tacn:`eauto` is a transitivity
+      of :tacn:`eauto` instead of the hint list of :tacn:`auto` and a warning is printed. A
+      typical hint that would only be used by :tacn:`eauto` is a transitivity
       lemma.
 
       .. exn:: @qualid cannot be used as a hint
 
          The head symbol of the type of :n:`@qualid` is a bound variable
          such that this tactic cannot be associated to a constant.
 
-   .. cmdv:: Hint Resolve {+ @qualid} : @ident
-
-      Adds each :n:`Hint Resolve @qualid`.
-
-   .. cmdv:: Hint Resolve -> @qualid : @ident
-
-      Adds the left-to-right implication of an equivalence as a hint (informally
-      the hint will be used as :n:`apply <- @qualid`, although as mentioned
-      before, the tactic actually used is a restricted version of
+      The second form adds the left-to-right (`->`) or right-ot-left implication (`<-`)
+      of an equivalence as a hint (informally
+      the hint will be used as, respectively, :tacn:`apply` :n:`-> @qualid` or
+      :n::tacn:`apply` :n:`<- @qualid`,
+      although as mentioned before, the tactic actually used is a restricted version of
       :tacn:`apply`).
```

Comment:
I would move the description of the second form above the error message (or rather make it into a description of `->` / `<-` in a description list as discussed above).